### PR TITLE
cleanup/prep endpoint semantics for #1558 and #1557

### PR DESCRIPTION
**Pull request by subtly**
Mostly semantic. Added test_allowLocal and friends.

### CODE REVIEW

File: libp2p/NodeTable.cpp [hunk 1]
```diff
@@ -81,12 +80,12 @@ shared_ptr<NodeEntry> NodeTable::addNode(Node const& _node)
...
 	{
-		clog(NodeTableConnect) << "Sending public key discovery Ping to" << _node.endpoint.udp << "(Advertising:" << m_node.endpoint.udp << ")";
+		clog(NodeTableConnect) << "Sending public key discovery Ping to" << (bi::udp::endpoint)_node.endpoint << "(Advertising:" << (bi::udp::endpoint)m_node.endpoint << ")";
 		{
 			Guard l(x_pubkDiscoverPings);
-			m_pubkDiscoverPings[_node.endpoint.udp.address()] = std::chrono::steady_clock::now();
+			m_pubkDiscoverPings[_node.endpoint.address] = std::chrono::steady_clock::now();
 		}
-		PingNode p(_node.endpoint.udp, m_node.endpoint.udp.address().to_string(), m_node.endpoint.udp.port());
+		PingNode p((bi::udp::endpoint)_node.endpoint, m_node.endpoint.address.to_string(), m_node.endpoint.udpPort);
```
Comment (subtly) [inline libp2p/NodeTable.cpp]: Cast can be removed.

File: test/net.cpp [hunk 2]
```diff
@@ -32,7 +32,13 @@ using namespace dev::p2p;
 namespace ba = boost::asio;
 namespace bi = ba::ip;
 
-BOOST_AUTO_TEST_SUITE(net)
+struct NetFixture
+{
+	NetFixture() { dev::p2p::NodeIPEndpoint::test_allowLocal = true;; }
```
Comment (gavofyork) [inline test/net.cpp]: `;;` -> `;`

File: test/peer.cpp [hunk 3]
```diff
@@ -28,7 +28,13 @@ using namespace std;
 using namespace dev;
 using namespace dev::p2p;
 
-BOOST_AUTO_TEST_SUITE(p2p)
+struct P2PFixture
+{
+	P2PFixture() { dev::p2p::NodeIPEndpoint::test_allowLocal = true;; }
```
Comment (gavofyork) [inline test/peer.cpp]: `;;` -> `;`

**Comment by gavofyork** [comment]
so what are the changes here?

- a few renames
- endpoint -> address, port, port
- ?

**Comment by subtly** [comment]
Semantic changes (simplifies code for future PRs, esp re: endpoints):
* a few renames
* tcp endpoint, udp endpoint -> address, port, port
* use of NodeIPEndpoint where possible
* added UnspecifiedNodeIPEndpoint and UnspecifiedNode w/tests (similar to EmptyTrie)
* Peer is constructed from Node

Functional changes:
* restore network will restore peers w/private IPs iff required

**Pull request merged**